### PR TITLE
Add troubleshooting doc

### PR DESCRIPTION
**Pull request by krish7919**

### CODE REVIEW

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        cpu: 200m
+        memory: 3.5G
+
+So, when the MongoDB cache occupies a memory greater than 3.5GB, it is
+terminated by the ``kubelet``.
+This can usually be verified by logging in to the worker node running MongoDB
+container and looking at the syslog (the ``journalctl`` command should usually
+work). It should show an error message as:
+
+TODO: paste the error message the next time this error happens!
```
Comment (krish7919) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: This is a placeholder till I get my hands on the precise stacktrace in kernel logs.

**Comment by ttmc** [comment]
Resolves #1514

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+-------------------
+
+We define the following in the ``mongo-ss.yaml`` file:
+
+.. code:: bash
+
+    resources:
+      limits:
+        cpu: 200m
+        memory: 3.5G
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: This has changed to 5G

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+So, when the MongoDB cache occupies a memory greater than 3.5GB, it is
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: Cut the "So," and change 3.5GB to 5GB

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+terminated by the ``kubelet``.
+This can usually be verified by logging in to the worker node running MongoDB
+container and looking at the syslog (the ``journalctl`` command should usually
+work). It should show an error message as:
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: "such as:"

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+The solution to this problem is to restart ``kube-proxy`` on the affected
+worker/agent node. Login to the worker node and run:
+
+.. code:: bash
+
+    docker stop `docker ps | grep k8s_kube-proxy | cut -d" " -f1`
+    
+    docker logs -f `docker ps | grep k8s_kube-proxy | cut -d" " -f1`
+
+This issue is fixed in Kubernetes v1.7.
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: Maybe give a link to an issue or pull request about the fix, if you can find one.

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+
+4. Single Disk Attached to Multiple Mountpoints in a Container
+--------------------------------------------------------------
+
+This is currently the issue faced in one of the clusters and being debugged by
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: Maybe say the date the issue happened and which cluster. You can leave it vague (e.g. just "testnet").

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+the support team at Microsoft.
+
+An issue has been logged `here
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: Replace "here" with "in the Azure/acs-engine repo on GitHub"

File: docs/server/source/production-deployment-template/troubleshoot.rst (hunk 3 above)
Comment (ttmc) [reply docs/server/source/production-deployment-template/troubleshoot.rst]: Maybe comment that it used to happen with 3.5GB but maybe the problem is gone now that it's 5GB.

File: docs/server/source/production-deployment-template/troubleshoot.rst [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    resources:
+      limits:
+        cpu: 200m
+        memory: 5G
+
+When the MongoDB cache occupies a memory greater than 5GB, it is
+terminated by the ``kubelet``.
+This can usually be verified by logging in to the worker node running MongoDB
+container and looking at the syslog (the ``journalctl`` command should usually
+work). It should show an error message such as:
```
Comment (ttmc) [inline docs/server/source/production-deployment-template/troubleshoot.rst]: Such as what? Maybe just remove the "It should show an error message such as:"

At a higher level, this issue never actually happened with the 5GB case, as far as I know. PR #1757 is what bumped it up to 5GB. Maybe say that (i.e. it happened when "memory: 3.5G" but not 5G).

**Pull request merged**